### ISSUE TITLE
Warnung unter PHP 8.1.16 bei aktivem Dev-Mode unter Contao 

### ISSUE THREAD

**Issue by Blog404DE** [heimrichhannot/contao-fieldpalette-bundle#19]
**Context**
Contao version: 4.13.16
Bundle version: 0.6.12
PHP version: 8.1.16

**Description**
Zuerst - während der Entwicklungs-Phase habe ich gerne Contao im Dev-Mode laufen um auch entsprechend Warnungen und deprecated-Meldungen zu erhalten.  

Im Zusammenhang mit der Migration zu PHP 8.1 (auf dem Server und PHP 8.2 auf der Entwicklungs-Umgebung) ist mir folgendes Verhalten im Zusammenhang mit der Erweiterung [netzhirsch/cookie-opt-in-bundle](https://github.com/Netzhirsch/CookieOptInBundle) aufgefallen:

Im Backend kommt es beim editieren des "Cookie Opt In"-Moduls zu folgenden Fehler:

```php
ErrorException:
Warning: Undefined array key "tl_layout"

  at vendor/contao/core-bundle/src/Resources/contao/widgets/ModuleWizard.php:214
  at Contao\ModuleWizard->generate()
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Widget.php:660)
  at Contao\Widget->generateWithError(true)
     (vendor/contao/core-bundle/src/Resources/contao/templates/backend/be_widget.html5:3)
  at include('/srv/beta.dutzi.info/vendor/contao/core-bundle/src/Resources/contao/templates/backend/be_widget.html5')
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/TemplateInheritance.php:108)
  at Contao\Widget->inherit()
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Widget.php:607)
  at Contao\Widget->parse()
     (vendor/contao/core-bundle/src/Resources/contao/classes/DataContainer.php:795)
  at Contao\DataContainer->row('	name	,type	;headlineCookieOptInBar	,questionHint	;saveButton	,saveAllButton	,highlightSaveAllButton	,rejectAllButton	;infoButtonShow	,infoButtonHide	,infoHint	;align	,space	,cssID	;cookieGroups	,cookieTools	,otherScripts	,i_frame_video	,i_frame_maps	,i_frame_i_frame	,i_frame_always_load	,i_frame_load	;expireTime	;privacyPolicy	,imprint	,excludePages	;respectDoNotTrack	,optOut	;zIndex	,blockSite	;templateBar	,defaultCss	,cssTemplateStyle	,maxWidth	,position	,animation	;ipFormatSave	;isNewCookieVersion	;languageSwitch')
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:1972)
  at Contao\DC_Table->edit()
     (vendor/contao/core-bundle/src/Resources/contao/classes/Backend.php:667)
  at Contao\Backend->getBackendModule('themes', null)
     (vendor/contao/core-bundle/src/Resources/contao/controllers/BackendMain.php:168)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:49)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:163)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:75)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:44)
  at require('/srv/beta.dutzi.info/web/index.php')
     (web/app.php:13)
``` 

Eine kurze Prüfung zeigte, dass die entsprechende Sprach-Datei von Contao geladen wurde. Der Fehler kann man relativ einfach lösen, indem man [an folgender Stelle](https://github.com/heimrichhannot/contao-fieldpalette-bundle/blob/8250efabfe12533d9702f1a86f87dc305df6f75d/src/Dca/DcaGenerator.php#L19) die entsprechende Sprachdatei läd:

```php
Controller::loadLanguageFile('tl_layout');
```
Ich denke es macht Sinn direkt an dieser Stelle zu laden und nicht in die Erweiterung zu verschieben, welche contao-fieldpalette-bundle einsetzt. 

Allerdings kommt es im Anschluss noch zu einem weiteren Fehler, bei dem ich mir (ohne mich mehr im Detail mit dem Code der Erweiterung zu beschäftigen) nicht ganz sicher bin, ob dies durch diese Erweiterung oder durch ein Fehler von _netzhirsch/cookie-opt-in-bundle_ ausgelöst wird:

```php
ErrorException:
Warning: Attempt to read property "cookieTools" on null

  at vendor/heimrichhannot/contao-fieldpalette-bundle/src/EventListener/Contao/ExecutePostActionsListener.php:53
  at HeimrichHannot\FieldpaletteBundle\EventListener\Contao\ExecutePostActionsListener->__invoke('refreshFieldPaletteField', object(DC_Table))
     (vendor/contao/core-bundle/src/Resources/contao/classes/Ajax.php:527)
  at Contao\Ajax->executePostActionsHook(object(DC_Table))
     (vendor/contao/core-bundle/src/Resources/contao/classes/Ajax.php:509)
  at Contao\Ajax->executePostActions(object(DC_Table))
     (vendor/contao/core-bundle/src/Resources/contao/classes/Backend.php:430)
  at Contao\Backend->getBackendModule('themes', null)
     (vendor/contao/core-bundle/src/Resources/contao/controllers/BackendMain.php:168)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:49)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:163)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:75)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:44)
  at require('/srv/beta.dutzi.info/web/index.php')
     (web/app.php:13)
```

Die Meldung trifft auf beim AJAX-Request, nachdem man innerhalb des oben erwähnten Backend-Moduls innerhalb der beiden vorhandenen Tabellen (z.B. "Tools") etwas ändert. Daher wäre meine Frage, ob dies ggf. am Bundle selber oder an _netzhirsch/cookie-opt-in-bundle_ liegen könnte?

**Comment by koertho** [comment]
Hallo, zuerst einmal danke fürs Reporten.

Beim ersten Fehler sehe ich leider keinerlei Zusammenhang mit diesem Bundle. ModuleWizard ist ein Core-Widget, ein Label von tl_layout wird in der genannten Funktion nicht angefordert. Wenn, dann lieger der Fehler im Netzhirsch-Opt-In-Bundle. Von daher macht es schon Sinn, die Sprachdatei dort zu laden, wenn nur dort benötigt wird.

Den zweiten Fehler konnte ich reproduzieren und er wird in der nächsten Version gefixt.

**Comment by Blog404DE** [comment]
Perfekt - vielen Dank 👍 

Bei dem erwähnten Bundle habe ich kurz ein Pull-Request eingereicht, welche die fehlende Sprach-Datei für das Backend-Modul lädt.
